planner/core/stats.go

```diff
@@ -178,19 +179,51 @@ func (ds *DataSource) DeriveStats(childStats []*property.StatsInfo) (*property.S
 		}
 	}
 	// Consider the IndexMergePath. Now, we just generate `IndexMergePath` in DNF case.
-	if len(ds.pushedDownConds) > 0 && len(ds.possibleAccessPaths) > 1 && ds.ctx.GetSessionVars().GetEnableIndexMerge() {
-		needConsiderIndexMerge := true
-		for i := 1; i < len(ds.possibleAccessPaths); i++ {
-			if len(ds.possibleAccessPaths[i].accessConds) != 0 {
-				needConsiderIndexMerge = false
-				break
+	if len(ds.pushedDownConds) > 0 && len(ds.possibleAccessPaths) > 1 &&
+		ds.ctx.GetSessionVars().GetEnableIndexMerge() {
```

Comment:
I think hint should have higer priorities than session variable, that is even if it is disabled by session vars, hint should also take effects.